### PR TITLE
Remove outputs from `file-system` mods that dup `network_storage`

### PR DESCRIPTION
**Pull request by mr0re1**

### CODE REVIEW

**Comment by cboneti** [comment]
I would like Nick to review this as he was the last person to take a closer look at the storage modules design.

**Comment by mr0re1** [comment]
All integration tests (except PR-validation) have passed before latest push

File: modules/scripts/startup-script/README.md [hunk 1]
```diff
@@ -138,7 +138,7 @@ sudo journalctl -u google-startup-scripts.service
   settings:
     runners:
       # Some modules such as filestore have runners as outputs for convenience:
-      - $(homefs.install_nfs_client_runner)
```
Comment (nick-stroud) [inline modules/scripts/startup-script/README.md:141]: I am not convinced that trading
```
$(homefs.install_nfs_client_runner)

for

((module.homefs.network_storage.client_install_runner))
```

is better. Is there any way we could use `$(homefs.network_storage.client_install_runner)`. We generally not favored users using the `(())` syntax. Can you add some justification as to why this change is improving the user experience.

File: docs/network_storage.md
Comment (nick-stroud) [inline docs/network_storage.md:1]: The PR description says "Add unit test to ensure that all file-system mod ouptut network_storage."
1. I don't see any diffs including unit tests in this pr.
2. I had thought you already added that change in a different PR I reviewed.

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: docs/network_storage.md
Comment (mr0re1) [reply docs/network_storage.md:1]: Stale description, test where merged in as a separate PR. Updated description.

File: modules/scripts/startup-script/README.md (hunk 1 above)
Comment (mr0re1) [reply modules/scripts/startup-script/README.md:141]: Addressed by using newly supported $() functionality.

File: community/modules/file-system/cloud-storage-bucket/README.md [hunk 2]
```diff
@@ -80,8 +80,8 @@ following example:
     source: modules/scripts/startup-script
     settings:
       runners:
-      - $(bucket.client_install_runner)
-      - $(bucket.mount_runner)
+      - $(bucket.network_storage.client_install_runner)
+      - $(bucket.network_storage.mount_runner))
```
Comment (nick-stroud) [inline community/modules/file-system/cloud-storage-bucket/README.md:84]: Typo: extra parens.

File: community/modules/file-system/nfs-server/main.tf [hunk 3]
```diff
@@ -43,11 +43,6 @@ locals {
       "destination" = "mount${replace(mount, "/", "_")}.sh"
     }
   ]
-  ansible_mount_runner = {
-    "type"        = "ansible-local"
-    "source"      = "${path.module}/scripts/mount.yaml"
```
Comment (nick-stroud) [inline community/modules/file-system/nfs-server/main.tf:48]: If this is the only place it is used we should also delete the source file.

File: community/modules/file-system/nfs-server/README.md
Comment (nick-stroud) [inline community/modules/file-system/nfs-server/README.md:1]: I think this change is healthy but will probably break some users. I think we need to do more to help ease that pain. My recommendations:

- Update title to make deprecation clear and call out deprecated variables. Something like: 
  > Deprecated outputs: `client_install_runner`, `mount_runner` & related outputs removed in favor of `network_storage`
- Clearly document in description how to migrate from the old functionality to replacement functionality. This way if someone finds this PR when looking for why variables went missing they will have a clear path forward.
- Add a file to the repo named `docs/deprecations.md` and add a section in there describing the deprecation and how to migrate to the new format (can be same as the description). We can point to this documentation from release notes.

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: tools/cloud-build/daily-tests/blueprints/lustre-with-new-vpc.yaml [hunk 4]
```diff
@@ -49,8 +49,8 @@ deployment_groups:
     source: modules/scripts/startup-script
     settings:
       runners:
-      - $(scratchfs.install_ddn_lustre_client_runner)
-      - $(scratchfs.mount_runner)
+      - $(scratchfs.network_storage.client_install_runner)
```
Comment (nick-stroud) [inline tools/cloud-build/daily-tests/blueprints/lustre-with-new-vpc.yaml:52]: You can get rid of the use of startup script completely here and just use the file system downstream. They were kept here to test the runners that you are deleting. See #1429 for example.

**Review by nick-stroud — CHANGES_REQUESTED** [review]
